plaso/parsers/text_plugins/apple_pstxt.py

```diff
@@ -0,0 +1,289 @@
+      ParseError: if the structure cannot be parsed.
+    """
+    # Retrieve the data from the file's metadata
+    self._SetEstimatedDate(parser_mediator)
+
+    if key == 'log_line':
+      event_data = ApplePSTxtEventData()
+      event_data.user = self._GetValueFromStructure(structure, 'user')
+      event_data.user_identifier = self._GetValueFromStructure(
+        structure, 'user_identifier')
+      event_data.persona = self._GetValueFromStructure(structure, 'persona')
+      event_data.process_identifier = self._GetValueFromStructure(
+        structure, 'process_identifier')
+      event_data.parent_process_identifier = self._GetValueFromStructure(
+        structure, 'parent_process_identifier')
+      event_data.flags = self._GetValueFromStructure(structure, 'flags')
+      event_data.cpu = self._GetValueFromStructure(structure, 'cpu')
+      event_data.memory = self._GetValueFromStructure(structure, 'memory')
+      event_data.scheduling_priority = self._GetValueFromStructure(
+        structure, 'scheduling_priority')
+      event_data.nice_value = self._GetValueFromStructure(
+        structure, 'nice_value')
+      event_data.virtual_size = self._GetValueFromStructure(
+        structure, 'virtual_size')
+      event_data.resident_set_size = self._GetValueFromStructure(
+        structure, 'resident_set_size')
+      event_data.wait_channel = self._GetValueFromStructure(
+        structure, 'wait_channel')
+      event_data.control_terminal_name = self._GetValueFromStructure(
+        structure, 'control_terminal_name')
+      event_data.symbolic_process_state = self._GetValueFromStructure(
+        structure, 'symbolic_process_state')
+      event_data.up_time = self._GetValueFromStructure(structure, 'up_time')
+      event_data.command = self._GetValueFromStructure(structure, 'command')
+
+      start_time_string = self._GetValueFromStructure(
+        structure, 'start_time')
+      event_data.start_time = self._ParseStartTime(start_time_string)
+
+      parser_mediator.ProduceEventData(event_data)
+
+  def _ParseStartTime(self, start_time_string):
+    """Parses the start time element of a log line.
+
+    Args:
+      start_time_string (str): start time element of a log line.
+
+    Returns:
+      dfdatetime.TimeElements: date and time value.
+
+    Raises:
+      ParseError: if a valid date and time value cannot be derived from
+          the time elements.
+    """
+    try:
+      # Case where it's just a time
+      if (
+          start_time_string[0] in pyparsing.nums
+          and start_time_string[-1] == 'M'):
+        parsed_elements = self._TIME.parse_string(start_time_string)
+
+        hours = int(parsed_elements.get('hours'))
+        minutes = int(parsed_elements.get('minutes'))
+        ampm = parsed_elements.get('ampm')
+        if ampm == 'PM':
+          hours += 12
+        if hours == 24:
+          hours = 0
+
+        # Retrieve year, month, day from dateless helper
+        year = self._date[0]
```

Comment:
Based on the format I assume date here should be the most recent date. However the date-less log helper will do the opposite it will return the earliest date as its base.

Reply:
I disagree on that point. Since the file is created once by the ps command and should not be modified, the earliest timestamp should be the correct one. If the file is moved or accessed, then the most recent date would not be correct. For instance, the test file on my system as an earliest date of April 13, 2022 (modified) whereas the latest date is April 12, 2024 (accessed today).